core/src/lib/auth/internal/CoreStitchAuth.ts

```diff
@@ -159,11 +154,10 @@ export default abstract class CoreStitchAuth<TStitchUser extends CoreStitchUser>
     stitchReq: StitchAuthDocRequest
   ): Promise<Response> {
     const newReqBuilder = stitchReq.builder;
-    newReqBuilder.withBody(stitchReq.document);
-    const newHeaders = newReqBuilder.headers; // This is not a copy
+    newReqBuilder.withBody(JSON.stringify(stitchReq.document));
+    const newHeaders = newReqBuilder.headers || {}; // This is not a copy
```

Comment:
For example, now that `newReqBuilder` is typed, the TS compiler was able to catch the fact that the `newReqBuilder.headers` could be undefined, so I added this safeguard in a lot of places